src/entities/itemsiter.rs

```diff
@@ -0,0 +1,91 @@
+use page::Page;
+use serde::Deserialize;
+
+/// Abstracts away the `next_page` logic into a single stream of items
+///
+/// ```no_run
+/// # extern crate mammut;
+/// # use mammut::{Data, Mastodon};
+/// # use std::error::Error;
+/// # fn main() -> Result<(), Box<Error>> {
+/// # let data = Data {
+/// #   base: "".into(),
+/// #   client_id: "".into(),
+/// #   client_secret: "".into(),
+/// #   redirect: "".into(),
+/// #   token: "".into(),
+/// # };
+/// let client = Mastodon::from_data(data);
+/// let statuses = client.statuses("user-id", None)?;
+/// for status in statuses.items_iter() {
+///   // do something with `status`
+/// }
+/// # Ok(())
+/// # }
+/// ```
+pub struct ItemsIter<'a, T: Clone + for<'de> Deserialize<'de>> {
```

Comment:
Should this struct be public? If it's hidden in the return type should we be showing it?

Reply:
It needs to be `pub(crate)` so it can be accessed from `src/page.rs` but it doesn't have to be `pub`